lib/bulk/ordered.js

```diff
@@ -25,7 +25,11 @@ const isPromiseLike = require('../utils').isPromiseLike;
 function addToOperationsList(bulkOperation, docType, document) {
   // Get the bsonSize
   const bsonSize = bson.calculateObjectSize(document, {
-    checkKeys: false
+    checkKeys: false,
+
+    // Since we don't know what the user selected for BSON options here,
+    // err on the safe side, and check the size with ignoreUndefined: false.
+    ignoreUndefined: false
```

Comment:
I think the more complete solution will save off a `ignoreUndefined` in the base class for the bulk operation (with a default of `false`), and use that for calculations. Was there a reason you didn't want to support the option from the operation level?

Reply:
My reasoning was that a user can pass in `ignoreUndefined: true` to `initialize[Un]orderedBulkOp`, and then pass `ignoreUndefined: false` into `op.execute`. Edge case, but IMO we're better off playing it safe.